src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6559,6 +6632,12 @@ private BoundExpression BindArrayAccess(ExpressionSyntax node, BoundExpression e
             Debug.Assert(expr != null);
             Debug.Assert(arguments != null);
 
+            BoundExpression prototypeExtensionMethod = GetPrototypeExtensionIndexerOpt(node, expr, arguments, diagnostics);
```

Comment:
>prototypeExtensionMethod [](start = 28, length = 24)

Consider renaming since the result is not a method, it's a call to an indexer. Perhaps `indexerAccess`.